src/SchemaBuilder/Decorator/ClassDecorator/AnnotationClassSchemaDecorator.php

```diff
@@ -34,9 +35,9 @@ public function decorateClassSchemaBuilder(
         }
 
         $classAnnotations = $this->getClassAnnotations($classReflection);
-        /** @var ?OA\ComponentSchema $componentSchema */
-        $componentSchema = $this->findAnnotation($classAnnotations, OA\ComponentSchema::class, true);
-        if ($componentSchema && $componentSchema->schemaName) {
+
+        $componentSchema = $this->findAnnotation($classAnnotations, ComponentSchema::class, true);
+        if ($componentSchema instanceof ComponentSchema && $componentSchema->schemaName) {
             $builder = $builder->withSchemaName($componentSchema->schemaName);
```

Comment:
This check was updated in more places.
Before the `!== null` value was checked, now the PHPStorm and PHPStan sees exact type thanks to `instanceof`